frontend/src/scenes/insights/Insights.tsx

```diff
@@ -152,6 +157,40 @@ export function Insights(): JSX.Element {
         },
     })
 
+    // Empty states that completely replace the graph
+    const BlockingEmptyState = (() => {
+        // Insight specific empty states - note order is important here
+        if (activeView === ViewType.FUNNELS) {
+            if (!areFiltersValid) {
+                return <FunnelInvalidFiltersEmptyState />
+            }
+            if (!areExclusionFiltersValid) {
+                return <FunnelInvalidExclusionFiltersEmptyState />
+            }
+            if (!isValidFunnel && !(resultsLoading || isLoading)) {
+                return <FunnelEmptyState />
+            }
+        }
+
+        // Insight agnostic empty states
+        if (showErrorMessage) {
+            return <ErrorMessage />
+        }
+        if (showTimeoutMessage) {
+            return <TimeOut isLoading={isLoading} />
+        }
+
+        return null
+    })()
```

Comment:
I guess moving this here makes sense for insights, but in turn breaks dashboards. Here's for example something with just one step that I added to my dashboard. Somehow nothing is shown. Not sure about all the other error states, as dashboard items can also timeout, etc.

![image](https://user-images.githubusercontent.com/53387/130614967-6a415611-73a7-40d0-bade-3c049483061c.png)

Reply:
Nice catch—because these states can exist in both dashboards and insights, sharing these empty states across both situations should fix this.

Reply:
Fixed!

![Screen Shot 2021-08-24 at 10 30 37 AM](https://user-images.githubusercontent.com/13460330/130662651-c7d23153-2da6-45a2-8a15-ff9b284825ad.png)